tests/define_status_test_project.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+import flow
+from flow import FlowProject
+
+
+class _TestProject(FlowProject):
+    pass
+
+
+group1 = _TestProject.make_group(name="group1")
+group2 = _TestProject.make_group(name="group2", options="--num-passes=2")
+
+
+@_TestProject.label
+def default_label(job):
+    return True
+
+
+@_TestProject.label
+def negative_default_label(job):
+    return False
+
+
+@_TestProject.label("named_label")
+def anonymous_label(job):
+    return True
+
+
+@_TestProject.label
+def b_is_even(job):
+    try:
+        return job.sp.b % 2 == 0
+    except AttributeError:
+        return False
+
+
+@_TestProject.operation.with_directives(
+    {
+        # Explicitly set a "bad" directive that is unused by the template.
+        # The submit interface should warn about unused directives.
+        "bad_directive": 0,
+        # But not this one:
+        "np": 1,
+    }
+)
+@flow.cmd
+@_TestProject.pre(b_is_even)
+@group1
+@_TestProject.post.isfile("world.txt")
```

Comment:
```suggestion
@flow.cmd
@group1
@_TestProject.pre(b_is_even)
@_TestProject.post.isfile("world.txt")
```
reordering just for clarity